### PR TITLE
Do not reconcile created tasks

### PR DESCRIPTION
**Pull request by alenkacz**
Fixes a race condition where created tasks that have not yet been sent to Mesos were included in reconciliation

JIRA issues: MARATHON-8372

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala [hunk 1]
```diff
@@ -505,28 +504,10 @@ class SchedulerActions(
 object TaskStatusCollector {
   def collectTaskStatusFor(instances: Seq[Instance]): Seq[mesos.Protos.TaskStatus] = {
     instances.flatMap { instance =>
-      instance.tasksMap.values.withFilter(task => !task.isTerminal && !task.isReserved).map { task =>
-        task.status.mesosStatus.getOrElse(initialMesosStatusFor(task, instance.agentInfo))
+      instance.tasksMap.values.collect {
+        case task @ Task(_, _, Task.Status(_, _, Some(mesosStatus), _, _)) if !task.isTerminal =>
```
Comment (meichstedt) [inline src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Why did you choose to not filter out reserved anymore?

There's still two cases a task is in Reserved: (1) when it's just been created alongside the reservation, and (2) once we receive a terminal status update, which will transition the task back to `Condition.Reserved`. The former case is caught by leaving out tasks w/o status. The latter is not  contained in `isTerminal`. Since the task was reported terminal we don't want to reconcile for it – right?

Comment (meichstedt) [reply src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Even if this is probably temporary – can you please add a comment to the `def collectTaskStatusFor` which explains that the OverdueTasksActor is taking care of unconfirmed tasks?

File: src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala [hunk 2]
```diff
@@ -158,10 +157,10 @@ class MarathonSchedulerActorTest extends AkkaUnitTest with ImplicitSender with G
       expectMsg(TasksReconciled)
 
       val nonTerminalTasks = instance.tasksMap.values.filter(!_.task.isTerminal)
-      assert(nonTerminalTasks.size == 7, "We should have 7 non-terminal tasks")
+      assert(nonTerminalTasks.size == 6, "We should have 7 non-terminal tasks")
```
Comment (meichstedt) [inline src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala]: Why did you remove the `.addTaskCreated`? Wouldn't it be OK to leave it in and make sure it's not considered?

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala (hunk 1 above)
Comment (timcharper) [reply src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: Yeah, the fact that we use Reserved for tasks that are killed and tasks that haven't been launched yet is very confusing. Second this question about why it was removed.

File: src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala (hunk 2 above)
Comment (timcharper) [reply src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala]: ➕

File: src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala (hunk 1 above)
Comment (alenkacz) [reply src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala]: well when we land phase 2, that won't be the case anymore. But you're right, on master I cannot remove that :(

File: src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala (hunk 2 above)
Comment (alenkacz) [reply src/test/scala/mesosphere/marathon/MarathonSchedulerActorTest.scala]: because this test tests for nonterminal tasks (that they are not included). I added another test covering created.
